### PR TITLE
SoundWire: use HDAudio registers for LunarLake

### PR DESCRIPTION
**Pull request by plbossart**
important update reflecting updated programming flows, the use of SDWWAKEEN is reserved to the DSP and the global HDAudio registers must be used to detect wakes from low-power modes.

@bardliao @RanderWang comments welcome.

### CODE REVIEW

File: drivers/soundwire/intel_ace2x.c [hunk 1]
```diff
@@ -34,37 +36,56 @@ static void intel_shim_vs_init(struct sdw_intel *sdw)
...
 
 static void intel_shim_wake(struct sdw_intel *sdw, bool wake_enable)
 {
-	void __iomem *shim_vs = sdw->link_res->shim_vs;
+	u16 lsdiid = 0;
 	u16 wake_en;
 	u16 wake_sts;
+	int ret;
+
+	mutex_lock(sdw->link_res->shim_lock);
```
Comment (RanderWang) [inline drivers/soundwire/intel_ace2x.c:64]: shim_lock is used to protect sdw shim register access, but here we only access hda bus. should we use hbus->lock ?

File: drivers/soundwire/intel_auxdevice.c [hunk 2]
```diff
@@ -82,7 +82,9 @@ static void generic_new_peripheral_assigned(struct sdw_bus *bus,
 	}
 
 	if (sdw->link_res->hw_ops->program_sdi)
-		sdw->link_res->hw_ops->program_sdi(sdw, dev_num);
+		if (bus->dev_num_alloc == SDW_DEV_NUM_ALLOC_IDA_WAKE_ONLY &&
```
Comment (bardliao) [inline drivers/soundwire/intel_auxdevice.c]: Do we need to call program_sdi if dev_num_alloc = SDW_DEV_NUM_ALLOC_IDA, too?

File: drivers/soundwire/intel_ace2x.c (hunk 1 above)
Comment (plbossart) [reply drivers/soundwire/intel_ace2x.c:64]: good feedback but this is NOT only to protect the shim registers, which no longer exist in LNL. See https://github.com/thesofproject/linux/blob/topic/sof-dev/drivers/soundwire/intel_init.c#L78, this 'shim_lock' is actually the eml_lock.

File: drivers/soundwire/intel_auxdevice.c (hunk 2 above)
Comment (plbossart) [reply drivers/soundwire/intel_auxdevice.c]: we could, yes.

Comment (plbossart) [reply drivers/soundwire/intel_auxdevice.c]: done

**Review by bardliao — DISMISSED** [review]

**Review by RanderWang — DISMISSED** [review]

**Comment by plbossart** [comment]
This needs to be rebased once https://github.com/thesofproject/linux/pull/4440 is merged.

**Review by bardliao — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Pull request merged**